engine/src/main/resources/assets/shaders/chunk_frag.glsl

```diff
@@ -112,21 +112,17 @@ void main() {
     float invMax = inversesqrt(max(dot(tangent,tangent), dot(binormal,binormal)));
     mat3 tbn = mat3(tangent * invMax, binormal * invMax, normal);
 
+#if defined (PARALLAX_MAPPING)
     vec3 eyeTangentSpace = tbn * vertexViewPos.xyz;
 
     float height =  parallaxScale * texture2D(textureAtlasHeight, texCoord).r - parallaxBias;
 	texCoord += height * normalize(eyeTangentSpace).xy * TEXTURE_OFFSET;
 #endif
 
-    normalOpaque = (texture2D(textureAtlasNormal, texCoord).xyz * 2.0 - 1.0);
-
-    // Simplified tangent basis - because we can! Voxels and blocks are great
-    normalOpaque.xyz = vec3(worldSpaceNormal.x, normalOpaque.x, normalOpaque.y) * abs(worldSpaceNormal.xxx)
-        + vec3(normalOpaque.y, worldSpaceNormal.y, normalOpaque.x) * abs(worldSpaceNormal.yyy)
-        + vec3(normalOpaque.x, normalOpaque.y, worldSpaceNormal.z) * abs(worldSpaceNormal.zzz);
+    normalOpaque = normalize(texture2D(textureAtlasNormal, texCoord).xyz * 2.0 - 1.0);
+    normalOpaque = normalize(tbn * normalOpaque);
```

Comment:
This is the magic that allowed the normal mapping to rotate correctly as you viewed a block from different sides.